napari/utils/perf/_utils.py

```diff
@@ -81,29 +96,26 @@ def draw(self):
 
         def decorator(func):
             @functools.wraps(func)
-            def time_function(*args, **kwargs):
+            def wrapper(*args, **kwargs):
                 with perf_timer(timer_name, "decorator"):
                     return func(*args, **kwargs)
 
-            return time_function
+            return wrapper
 
         return decorator
 
 
 else:
-    # Disable both with hopefully zero run-time overhead.
-    if PYTHON_3_7:
-        perf_timer = contextlib.nullcontext()
-    else:
-
-        @contextlib.contextmanager
-        def perf_timer(name: str):
-            yield
+    # contextlib.nullcontext does not work with kwargs?
+    @contextlib.contextmanager
+    def perf_timer(name: str, category: Optional[str] = None, **kwargs):
+        yield
```

Comment:
Does this context ever get used? I thought if perfmon was off there were zero runtime modifications?

Reply:
I used to assign it to `contextlib.contextmanager` when disabled by I had to switch to this. I would hope Python could optimized away a generator that does nothing, but I'm not super optimistic about that, I'll do some timings.

Reply:
> But yeah if you leave a raw perf_timer in the code at all times

But we don't do this, right? I'd prefer to not enable this kind of pattern...

Reply:
There's no `perf_timers` checked in except one in the class `QApplicationWithTracing` but that is separately disabled. So there's no "active" `perf_timers` checked in as of this PR. The developer is expected to add them using the patcher. 

However there's two reasons you might want to temporarily add a `perf_timer` by hand while doing an intensive investigation of something:

1. You want to time a line of code or a block of code within a bigger function. And you don't (yet) want to refactor that code to extract that block into its own function that you can patch.

2. You want to add arguments to the timer. This is really for "tracing" not such much "timing". For example you could put the `shape` of an array as an argument to a `perf_timer` or the filename of what chunk you are loading, then in the chrome GUI you can see those values and that might help explain the timeline.

So I'd consider `perf_timers` like anything you'd add temporarily while debugging, like noisy print statement. Something you should remove before you commit or at least **before** you merge. This should be covered in the documentation I haven't written. Basically explain patching is "free" but manually added `perf_timers` are not completely free. I timed a "do nothing" generator as taking like a microsecond.

If there was a problem with people commiting and merging `perf_timers` we could write a CI step that checked for them, but I wouldn't bother unless it becomes a problem.